packages/core/test/types.test.ts

```diff
@@ -480,3 +482,17 @@ describe('interpreter', () => {
     });
   });
 });
+
+describe('spawn', () => {
+  it('spawned actor ref should be compatible with the result of ActorRefFrom', () => {
+    const createChild = () => createMachine({});
+
+    function createParent(_deps: {
+      spawnChild: () => ActorRefFrom<typeof createChild>;
+    }) {}
+
+    createParent({
+      spawnChild: () => spawn(createChild())
+    });
```

Comment:
I wish I could write a better test case for this, but unfortunately, within `createMachine` call TS gets lost when computing variance types and it allows what shouldn't be allowed. Since this is a known problem and there is a fix for it I've rechecked with an insiders build but it didn't end up erroring there either. I've reported this as a potential issue here: https://github.com/microsoft/TypeScript/pull/45628#issuecomment-1049714681